src/pages/City/CityPage.tsx

```diff
@@ -1,38 +1,27 @@
 import Weather from 'api/services/Weather';
 import { useMemo } from 'react';
-import { NavigateOptions, useLocation, useParams } from 'react-router-dom';
+import { useParams } from 'react-router-dom';
 import { convertSpecialCharacters } from 'utils/convertSpecialCharacters';
 import { PageContentContainer } from 'components/PageContentContainer';
 import { WidgetWrapper } from 'components/WidgetWrapper';
-import { CityName } from './CityPage.styled';
 import { CityPageTestIds } from './CityPageTestIds';
-import { CityNameLoader } from './components/CityNameLoader';
+import { CityNameWidget } from './components/CityNameWidget';
+import { NearestCitiesWeatherWidget } from './components/NearestCitiesWeatherWidget';
 
 type Params = {
   city: string;
 };
 
 const CityPage = () => {
   const { city } = useParams() as Params;
-  const normalizedCity = useMemo(()=> 
-    convertSpecialCharacters(city), [city]
-  );
+  const normalizedCity = useMemo(() => convertSpecialCharacters(city), [city]);
   const { data, isLoading } = Weather.useLocation(normalizedCity);
-  const { state }: NavigateOptions = useLocation();
-
-  const renderCityName = () => {
-    if (isLoading) {
-      return <CityNameLoader />;
-    }
-    if (data) {
-      return <CityName>{state?.cityName || data.location.name}</CityName>;
-    }
-  };
 
   return (
     <PageContentContainer data-testid={CityPageTestIds.Container}>
-      {renderCityName()}
+      <CityNameWidget loading={isLoading} cityName={data !== undefined ? data.location.name : undefined} />
       <WidgetWrapper data={data} isLoading={isLoading} data-testid={CityPageTestIds.Widgets} />
+      <NearestCitiesWeatherWidget loading={isLoading} cityName={data !== undefined ? data.location.name : undefined} />
```

Comment:
**suggestion** i belive `cityName={data?.location.name}` works the same and look cleaner

Reply:
Or `cityName={data && data.location.name}` will it work ? :D